src/components/MonacoEditor/SourceCodeHeading.tsx

```diff
@@ -183,51 +218,81 @@ const SourceCodeHeading: React.FC<IProps> = ({
+                    onClick={e => handleShowOptions(e)}
+                  />
+                )}
+              </span>
+            )}
+          </>
+        ) : (
+          <span
+            className={classes.monacoEditorTitle}
+            style={{ border: `1px solid ${color.themeBlue}` }}>
+            <input
+              onKeyDown={handleRenameTitleInputKeydown}
+              onChange={handleRenameTitleChange}
+              className={classes.monacoEditorRenameTitleInput}
+              value={renameTitleValue}
+              autoFocus
+            />
+            <CloseIcon
+              className={classes.commentTitleMenuIcon}
+              onClick={closeRenameTitle}
+              style={{
+                fontSize: 14,
+              }}
+            />
+          </span>
+        )}
+      </>
+    );
+  }
 
   return (
     <div className={classes.monacoEditorTitleHead}>
-      {!isRenameTitle ? (
-        <>
-          {(!isFetchingSourcecode || getSourceCodeIdFromUrl()) && (
-            <span className={classes.monacoEditorTitle}>
-              <span style={{ fontSize: 14, padding: 5 }}>
-                {!!sourceCodeTitle
-                  ? `${sourceCodeTitle}.js`
-                  : getSourceCodeIdFromUrl()
-                  ? ''
-                  : 'Untitled.js'}
-              </span>
-              {(isOwner || !getSourceCodeIdFromUrl() || (!user === true && !ownerId === true)) && (
-                <MoreVertIcon
-                  className={`${classes.commentTitleMenuIcon} comment__hide-title-menu-icon`}
-                  onClick={e => handleShowOptions(e)}
-                />
-              )}
-            </span>
-          )}
-        </>
-      ) : (
-        <span
-          className={classes.monacoEditorTitle}
-          style={{ border: `1px solid ${color.themeBlue}` }}>
-          <input
-            onKeyDown={handleRenameTitleInputKeydown}
-            onChange={handleRenameTitleChange}
-            className={classes.monacoEditorRenameTitleInput}
-            value={renameTitleValue}
-            autoFocus
-          />
-          <CloseIcon
-            className={classes.commentTitleMenuIcon}
-            onClick={closeRenameTitle}
-            style={{
-              fontSize: 14,
-            }}
-          />
-        </span>
-      )}
-      {renderTitleMenuOptions()}
+      <div style={{ display: 'flex', flex: 0.9, height: '100%' }}>
+        {renderSourcecodeTitle()}
+        {renderTitleMenuOptions()}
+      </div>
+      <div className={`${classes.createSourcecode} ${commonCss.flexRow}`}>
+        {!!ownerId && !isOwner && (
+          <Tooltip title="Fork project" leaveDelay={100} placement="bottom" enterDelay={100}>
+            <IconButton
+              onClick={forkSourcecode}
+              color="secondary"
+              classes={{ root: classes.createSourcecodeButton }}>
+              <DeviceHubIcon className={classes.createSourcecodeIcon} />
```

Comment:
I think it's better to call this ForkIcon. Sounds more intuitive.

Reply:
cool then, i will effect the change